plover/key_combo.py

```diff
@@ -0,0 +1,68 @@
+
+import re
+
+
+_SPLIT_RX = re.compile(r'(\s+|(?:\w+(?:\s*\()?)|.)')
+
+def parse_key_combo(combo_string, key_name_to_key_code=None):
+
+    if key_name_to_key_code is None:
+        key_name_to_key_code = lambda key_name: key_name
+
+    key_events = []
+    down_keys = []
+    token = None
+    count = 0
+
+    def _raise_error(exception, details):
+        msg = '%s:\n%s\n%s' % (
+            details, combo_string,
+            ' ' * count + '^' * len(token or ' '),
```

Comment:
I think underlining with `^^^^` is clear, but my experience with clang and similar leads me to expect the first character to be `^` and the rest to be `~`, so `^~~~` to point at the initial character and then underline a source range.

As you pointed out, neither works well if the output is set in a proportional font. To accommodate that, we'd probably need to do something like boxing the bogus token with `[…]` or linebreaking around the bogus token.

If we could trace the bogus key combo back to a specific dictionary file and line, that would also be font-independent, and we could include that info in the error message, too.